server/start.go

```diff
@@ -363,17 +365,39 @@ func startInProcess(ctx *server.Context, clientCtx client.Context, appCreator ty
 		}
 	}
 
-	var apiSrv *api.Server
-	if config.API.Enable {
+	if config.API.Enable || config.JSONRPC.Enable {
 		genDoc, err := genDocProvider()
 		if err != nil {
 			return err
 		}
 
-		clientCtx := clientCtx.
+		clientCtx = clientCtx.
 			WithHomeDir(home).
 			WithChainID(genDoc.ChainID)
 
+		// Set `GRPCClient` to `clientCtx` to enjoy concurrent grpc query.
+		if config.GRPC.Enable {
+			_, port, err := net.SplitHostPort(config.GRPC.Address)
+			if err != nil {
+				return err
+			}
+			grpcAddress := fmt.Sprintf("127.0.0.1:%s", port)
```

Comment:
why is this hard coded?

Reply:
Because it's `startInProcess`? I took it from SDK itself: https://github.com/cosmos/cosmos-sdk/blob/main/server/start.go#L368

Reply:
I see. What about the `maxRecvMsgSize` and `maxSendMsgSize`? Why are they not included?

Reply:
I copied from the [original PR](https://github.com/cosmos/cosmos-sdk/pull/11234/files#diff-97289af2ab66016f168df6fccac70626d125f170e5cf0841fcdc8fa624999608R301), I guess they are added later, hold on.

Reply:
now it's better copied.